keel-core/src/main/kotlin/com/netflix/spinnaker/keel/verification/VerificationRunner.kt

```diff
@@ -44,6 +52,35 @@ class VerificationRunner(
     }
   }
 
+  private fun anyPendingVerificationsForPreviousVersions(context: VerificationContext) =
+    verificationRepository
+      .pendingInEnvironment(context.deliveryConfig, context.environmentName)
+      // only consider other versions, we'll handle verifications for the version in context later
+      .filterNot { it.context.version == context.version }
+      // get the latest status by re-evaluating each one (which will update in the database)
+      .associateWith { it.context.latestStatus(it.verification) }
+      // we're only interested in any verifications that are actually running, if the last step has
+      // found they are now complete, or if they never even started, that should not stop us from
+      // launching new verifications
+      .filter { (_, status) -> status == PENDING }
```

Comment:
This is the crucial bit I changed. Previously I was looking at any status that was incomplete, but we may have old verifications that never got launched for whatever reason. We really only care about ones that are actually running.